src/bytes.rs

```diff
@@ -51,16 +57,40 @@ const _: () = {
 };
 
 extern "C" fn noop(_: *const (), _: usize) {}
+
 impl<'a> Bytes<'a> {
     /// Constructs an empty slice.
     ///
     /// Since [`Bytes`] are immutable slices, this mostly serves to create default values.
     pub const fn empty() -> Self {
         Self::from_static([].as_slice())
     }
+    const fn is_interned(&self) -> bool {
+        (unsafe {
+            (self as *const Self)
+                .cast::<u8>()
+                .add(core::mem::size_of::<Self>() - 1)
+                .read()
+        } & 1)
+            != 0
+    }
+    fn vtable(&self) -> Option<&'a BytesVt> {
+        if self.is_interned() {
+            None
+        } else {
+            unsafe { Some(self.vtable.as_ref()) }
+        }
```

Comment:
A style alternative would be to use:

```suggestion
        self.is_interned().not().then(|| unsafe { self.vtable.as_ref() })
```

Reply:
Indeed, I was trying to get this to be const, I actually just found a thing that might do it. If it doesn't I'll switch to the nicer style